src/PostgREST/AppState.hs

```diff
@@ -111,10 +113,17 @@ destroy = destroyPool
 
 initPool :: AppConfig -> IO SQL.Pool
 initPool AppConfig{..} =
-  SQL.acquire configDbPoolSize timeoutMicroseconds $ toUtf8 configDbUri
+  SQL.acquire configDbPoolSize timeoutMicroseconds configDbUriAndAppName
   where
     timeoutMicroseconds = (* oneSecond) <$> configDbPoolAcquisitionTimeout
     oneSecond = 1000000
+    configDbUriAndAppName = toUtf8 configDbUri <>
+      case uriQuery <$> parseURI (toS configDbUri) of
+        Nothing  -> " " <> appNameParam
```

Comment:
Testing this for Keyword/Value connection strings. Not fail proof, though. E.g. this string works and connects to the database, ignoring the `?` in password:

```
"postgres://authenticator:mysecr?????etpassword@localhost:5432/postgres?"
```

But `Network.URI` does not recognize it as a valid URI.

Reply:
Hmm, it's a bit hard to follow. Maybe we can have some doctests for this function?